src/main/java/com/boomi/flow/services/boomi/mdh/records/GoldenRecordRepository.java

```diff
@@ -159,13 +159,16 @@ public void delete(ApplicationConfiguration configuration, String universeId, Li
     }
 
     private List<MObject> update(ApplicationConfiguration configuration, List<MObject> objects, String universeId, String operation) {
-        var universe = client.findUniverse(configuration.getAtomHostname(), configuration.getAtomUsername(), configuration.getAtomPassword(), universeId);
-
-        objects.stream()
-                .filter(object -> Strings.isNullOrEmpty(object.getExternalId()))
-                .forEach(object -> Entities.AddRandomUniqueId(object, universe.getIdField()));
+        var universe = client.findUniverse(configuration.getHubHostname(), configuration.getHubUsername(), configuration.getHubToken(), universeId);
 
         var objectsBySource = objects.stream()
+                .map(object -> {
+                    if (Strings.isNullOrEmpty(object.getExternalId())) {
```

Comment:
I feel like `addRandomUniqueId` should just handle this `if` condition itself, as I've only seen it wrapped with this condition and never without it. Moving the condition to the method would also mean this isn't a multi-line lambda